core/src/main/java/org/apache/iceberg/metrics/CounterResultParser.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import org.apache.iceberg.metrics.MetricsContext.Unit;
+import org.apache.iceberg.metrics.ScanReport.CounterResult;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+class CounterResultParser {
+
+  private static final String MISSING_FIELD_ERROR_MSG =
+      "Cannot parse counter from '%s': Missing field '%s'";
+
+  private static final String NAME = "name";
+  private static final String UNIT = "unit";
+  private static final String VALUE = "value";
+
+  private CounterResultParser() {}
+
+  static String toJson(CounterResult counter) {
+    return toJson(counter, false);
+  }
+
+  static String toJson(CounterResult counter, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(counter, gen), pretty);
+  }
+
+  static void toJson(CounterResult counter, JsonGenerator gen) throws IOException {
+    toJson(counter, gen, true);
+  }
+
+  static void toJson(CounterResult counter, JsonGenerator gen, boolean includeName)
+      throws IOException {
+    Preconditions.checkArgument(null != counter, "Invalid counter: null");
+
+    gen.writeStartObject();
+    // ScanMetricsResultParser mainly uses this with includeName=false, since it includes the name
+    // in a parent structure
+    if (includeName) {
+      gen.writeStringField(NAME, counter.name());
+    }
```

Comment:
I don't think we should make this an option. Let's just omit name for now. I'm also skeptical that a counter should be created with its own name. I'd just keep it simple.

Reply:
I don't think we can omit the name. Metric frameworks (https://micrometer.io/docs/concepts#_meters) use the name to uniquely identify Counters/Timers. The flag itself only exists here so that we don't repeat the counter/timer name when they are part of `ScanMetricsResult` as can be seen in https://github.com/apache/iceberg/pull/5427/files#diff-31983148600c3f78b9628dc4d00e7928097ccb691788e1ae787f9e5870310712R164-R218